src/rgw/rgw_op.h

```diff
@@ -1704,9 +1709,10 @@ static inline void rgw_get_request_metadata(CephContext *cct,
 } /* rgw_get_request_metadata */
 
 static inline void encode_delete_at_attr(ceph::real_time delete_at,
-					map<string, bufferlist>& attrs)
+					map<string, bufferlist>& attrs, bool set_delete_at = false)
 {
-  if (real_clock::is_zero(delete_at)) {
+  map<string, bufferlist>::iterator iter = attrs.find(RGW_ATTR_DELETE_AT);
+  if (!set_delete_at && iter != attrs.end()) {
```

Comment:
What do you think about switching to `boost::optional<ceph::real_time>` instead of introducing the`set_delete_at` flag? Optional seems to be intended for handling cases exactly like this one.